sdk/ml/azure-ai-ml/azure/ai/ml/entities/_assets/environment.py

```diff
@@ -298,21 +298,23 @@ def validate(self):
             log_and_raise_error(err)
 
     def __eq__(self, other) -> bool:
-        return (
-            self.name == other.name
-            and self.id == other.id
-            and self.version == other.version
-            and self.description == other.description
-            and self.tags == other.tags
-            and self.properties == other.properties
-            and self.base_path == other.base_path
-            and self.image == other.image
-            and self.build == other.build
-            and self.conda_file == other.conda_file
-            and self.inference_config == other.inference_config
-            and self._is_anonymous == other._is_anonymous
-            and self.os_type == other.os_type
-        )
+        if isinstance(other, Environment):
+            return (
+                self.name == other.name
+                and self.id == other.id
+                and self.version == other.version
+                and self.description == other.description
+                and self.tags == other.tags
+                and self.properties == other.properties
+                and self.base_path == other.base_path
+                and self.image == other.image
+                and self.build == other.build
+                and self.conda_file == other.conda_file
+                and self.inference_config == other.inference_config
+                and self._is_anonymous == other._is_anonymous
+                and self.os_type == other.os_type
+            )
+        return False
```

Comment:
```suggestion
        if not isinstance(other, Environment):
            return NotImplemented
        return (
            self.name == other.name
            and self.id == other.id
            and self.version == other.version
            and self.description == other.description
            and self.tags == other.tags
            and self.properties == other.properties
            and self.base_path == other.base_path
            and self.image == other.image
            and self.build == other.build
            and self.conda_file == other.conda_file
            and self.inference_config == other.inference_config
            and self._is_anonymous == other._is_anonymous
            and self.os_type == other.os_type
        )
```

https://stackoverflow.com/questions/878943/why-return-notimplemented-instead-of-raising-notimplementederror

Reply:
Done